app/controllers/batch_controller.rb

```diff
@@ -0,0 +1,37 @@
+class BatchController < ApplicationController
+  def create
+    uris = payload.fetch(:uris)
+    checked_within = (payload[:checked_within] || 24.hours).to_i
+    callback_uri = payload[:callback_uri]
+
+    return render json: { error: { message: "No URIs given." } }, status: 400 if uris.empty?
+
+    links = uris.map do |uri|
+      Link.find_or_create_by(uri: uri)
+    end
+
+    checks = links.map do |link|
+      check = link.find_completed_check(within: checked_within)
+      check ? check : Check.create!(link: link)
+    end
```

Comment:
both of these loops feel like they'd make a bunch of queries when single ones could do most of it. 

I'd consider extracting it to a class - maybe `Batch::CreateChecks` and you could create a batch object first (I expect it'd be useful to store callback and checked_within values on it) which handles it? 

might also want to wrap the batch stuff in a transaction and see if we can do what we can to limit dangers of two requests at same time